src/lib/Microsoft.Health.Fhir.Ingest/Service/EventProcessingMeter.cs

```diff
@@ -19,21 +20,23 @@ public Task<EventStats> CalculateEventStats(IEnumerable<EventData> events)
 
             foreach (var e in events)
             {
-                var bodySizeBytes = e.Body.Array.Length;
-                ingressSizeBytes = ingressSizeBytes + bodySizeBytes + CalculateDictionarySizeBytes(e.Properties) + CalculateDictionarySizeBytes(e.SystemProperties);
+                var bodySizeBytes = e.Body.Length;
+                var propSizeBytes = CalculateDictionarySizeBytes(e.Properties);
+                var syspSizeBytes = CalculateDictionarySizeBytes(e.SystemProperties);
+                ingressSizeBytes += bodySizeBytes + propSizeBytes + syspSizeBytes;
             }
 
             var eventStats = new EventStats()
             {
-                TotalEventsProcessedBytes = ingressSizeBytes,
+                TotalEventsProcessedBytes = Convert.ToDouble(ingressSizeBytes),
             };
 
             return Task.FromResult(eventStats);
         }
 
-        private double CalculateDictionarySizeBytes(IDictionary<string, object> dictionary)
+        public static int CalculateDictionarySizeBytes(IEnumerable<KeyValuePair<string, object>> dictionary)
```

Comment:
Will we lose precision by moving from a `double` to an `int`? Should we change this to a `long`?